src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Calendar/Calendar.shared.cs

```diff
@@ -0,0 +1,653 @@
+		/// </summary>
+		public static readonly BindableProperty SelectionModeProperty =
+			BindableProperty.Create(nameof(SelectionMode), typeof(CalendarSelectionMode), typeof(Calendar), CalendarSelectionMode.SingleSelect, propertyChanged: OnModeChanged);
+
+		/// <summary>
+		/// Gets or sets the mode, which determines how <see cref="Calendar"/> handles days selection.
+		/// </summary>
+		public CalendarSelectionMode SelectionMode
+		{
+			get => (CalendarSelectionMode)GetValue(SelectionModeProperty);
+			set => SetValue(SelectionModeProperty, value);
+		}
+
+		static void OnModeChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			var calendar = (Calendar)bindable;
+			calendar.SelectedDays = Enumerable.Empty<DateTimeOffset>().ToList();
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="Calendar"/> class.
+		/// </summary>
+		public Calendar()
+		{
+			var layoutRoot = new Grid
+			{
+				Padding = 0,
+				ColumnSpacing = 0,
+				RowSpacing = 0
+			};
+
+			layoutRoot.RowDefinitions.Add(new RowDefinition { Height = GridLength.Auto });
+			layoutRoot.RowDefinitions.Add(new RowDefinition { Height = GridLength.Auto });
+
+			var gridWeekDayHeaders = new Grid();
+			gridWeekDayHeaders.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridWeekDayHeaders.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridWeekDayHeaders.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridWeekDayHeaders.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridWeekDayHeaders.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridWeekDayHeaders.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridWeekDayHeaders.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+
+			Grid.SetRow(gridWeekDayHeaders, 0);
+			layoutRoot.Children.Add(gridWeekDayHeaders);
+			this.gridWeekDayHeaders = gridWeekDayHeaders;
+
+			var gridDays = new Grid
+			{
+				ColumnSpacing = 0
+			};
+
+			gridDays.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridDays.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridDays.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridDays.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridDays.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridDays.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+			gridDays.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
+
+			Grid.SetRow(gridDays, 1);
+			layoutRoot.Children.Add(gridDays);
+			this.gridDays = gridDays;
+
+			Content = layoutRoot;
+		}
+
+		/// <summary>
+		/// Initializes <see cref="Calendar"/>.
+		/// </summary>
+		public void Initialize()
```

Comment:
Can we do it in another way? I mean, we would like to not have the users call `Init` methods at all. Our goal is to have more of a plug-and-play lib. I imagine that's because of the change to `Task` in other methods, right? If yes, did you try to disable the iOS animation while performing the month change?

Reply:
There was following issue: properties ```ShowWeekends``` and ```ShowDaysFromOtherMonths``` both have default value set to ```true```. When I set those values to ```false``` in XAML then methods ```OnShowDaysFromOtherMonthsPropertyChanged``` and ```OnShowWeekendsChanged``` are invoked. Both methods invoke ```UpdateCalendarDaysAsync```, which updates calendar days. Method ```UpdateCalendarDaysAsync``` was called twice when user was entering page where Calendar was used - this was hitting performance.

```Initialize``` method is a way to invoke ```UpdateCalendarDaysAsync``` only once when all other bindable properties are already set.

Let me know what you think about this.